src/sinks/pulsar/sink.rs

```diff
@@ -124,7 +125,23 @@ impl PulsarSink {
             .into_driver(service)
             .protocol("tcp");
 
-        sink.run().await
+        let result = sink.run().await;
+
+        debug!(message = "Receiver exhausted, closing Pulsar producer.");
```

Comment:
```suggestion
        debug!(message = "Receiver exhausted, closing Pulsar producers for each topic.");
```

This conveys the same message as was in the comment below I think